lib/object-path-immutable-transaction.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+const _assign = require('lodash.assign');
+const objectPath = require('object-path');
+const _isArray = require('lodash.isarray');
+const _isPlainObject = require('lodash.isplainobject');
+
+function ensureCloned(path, source, dest) {
+	if (!_isArray(path)) {
+		path = ('' + path).split('.');
+	}
+
+	// ensure all ancestor path segments are cloned
+	for (let i = 0; i < path.length; i++) {
+		if (dest[path[i]] === source[path[i]]) {
```

Comment:
I think this whole for block would be easier to read if you variableize `path[i]`.